From da3c83dbf0138cb1c26de35e69676a215dcef78d Mon Sep 17 00:00:00 2001
From: Nicolas Belin <nbelin@baylibre.com>
Date: Mon, 25 Mar 2019 14:58:23 +0100
Subject: [PATCH] Vesper: Using vesper config from alsa as the default source.

In order to use vesper in pulse audio, it is required to manually
load the alsa vesper module and set it as default.

Signed-off-by: Nicolas Belin <nbelin@baylibre.com>

---
 src/daemon/daemon.conf.in | 2 +-
 src/daemon/default.pa.in  | 2 +-
 src/pulse/client.conf.in  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/daemon/daemon.conf.in b/src/daemon/daemon.conf.in
index a955523..97952d1 100644
--- a/src/daemon/daemon.conf.in
+++ b/src/daemon/daemon.conf.in
@@ -83,7 +83,7 @@ ifelse(@HAVE_SYS_RESOURCE_H@, 1, [dnl
 ; default-sample-format = s16le
 ; default-sample-rate = 44100
 ; alternate-sample-rate = 48000
-; default-sample-channels = 2
+default-sample-channels = 8
 ; default-channel-map = front-left,front-right
 
 ; default-fragments = 4
diff --git a/src/daemon/default.pa.in b/src/daemon/default.pa.in
index a47486f..1529675 100755
--- a/src/daemon/default.pa.in
+++ b/src/daemon/default.pa.in
@@ -39,7 +39,7 @@ load-module module-switch-on-port-available
 ifelse(@HAVE_ALSA@, 1, [dnl
 #load-module module-alsa-sink
 load-module module-alsa-sink device=hw:0,4
-#load-module module-alsa-source device=hw:1,0
+load-module module-alsa-source device=vesper
 ])dnl
 ifelse(@HAVE_OSS_OUTPUT@, 1, [dnl
 #load-module module-oss device="/dev/dsp" sink_name=output source_name=input
diff --git a/src/pulse/client.conf.in b/src/pulse/client.conf.in
index ee094ca..b355251 100644
--- a/src/pulse/client.conf.in
+++ b/src/pulse/client.conf.in
@@ -18,7 +18,7 @@
 ## commenting.
 
 default-sink = alsa_output.hw_0_4
-; default-source =
+default-source = alsa_input.vesper
 ; default-server =
 ; default-dbus-server =
 
